core/threaded/reactor_threaded.c

```diff
@@ -937,6 +938,10 @@ bool _lf_worker_handle_STP_violation_for_reaction(int worker_number, reaction_t*
             
             // Reset the is_STP_violated because it has been dealt with
             reaction->is_STP_violated = false;
+        } else {
+        	// The intended tag cannot be respected and there is no handler.
+        	// Print an error message and return true.
+        	lf_print_error("Safe-to-process violation occurred! Discarding message.");
```

Comment:
Should this say something about not having an STP violation handler?

This is not discarding a message entirely, just not invoking this particular reaction.
If another reaction is triggered by the same tardy event but has an STP violation handler, there will not be an error message.